auto-instrumentation/httpurlconnection/library/src/main/java/io/opentelemetry/instrumentation/library/httpurlconnection/HttpUrlReplacements.java

```diff
@@ -0,0 +1,466 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.library.httpurlconnection;
+
+import static io.opentelemetry.instrumentation.library.httpurlconnection.internal.HttpUrlConnectionSingletons.instrumenter;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.library.httpurlconnection.internal.RequestPropertySetter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URLConnection;
+import java.util.List;
+import java.util.Map;
+import java.util.WeakHashMap;
+
+public class HttpUrlReplacements {
```

Comment:
This class is pretty huge and has a lot of internal duplication. Is it possible to do something like https://gist.github.com/breedx-splk/28af92b4d26421a0f0e2014fc76aa28b to help mitigate some of this? 

_Note: The inner interfaces are only required to maintain API 21 compat due to `Supplier` being API 24+._

Reply:
That looks great! Thanks for suggesting this! :) I will make the change, test and submit a commit shortly.